esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -1840,13 +1842,11 @@ void CFileSprayEx::readAndCheckSpraySourceReq(MemoryBuffer& srcxml, const char*
                 sourcePath.append(srcPath).trim();
             }
             const char * hostGroup = dropZone->queryProp("@hostGroup");
-            const char * host = "127.0.0.1";
+            const char * host = "localhost"; // by default assume storage plane mounted on dfuserver
             if (hostGroup)
             {
-               VStringBuffer xpath("storage/hostGroups[@name='%s']", srcPlane);
-               IPropertyTree * match = getGlobalConfigSP()->queryPropTree(xpath);
-               if (match)
-                   host = match->queryProp("hosts[1]");
+                Owned<IPropertyTree> match = getHostGroup(hostGroup,true);
+                host = match->queryProp("hosts[1]");
```

Comment:
Theoretically this could be freed when the scope is left.  Either assign in the block or clone when saving.

Same on other call below.